examples/read_nodes.rs

```diff
@@ -0,0 +1,115 @@
+// example that separates logic for reading different top-level nodes of xml tree
+
+use quick_xml::events::{BytesStart, Event};
+use quick_xml::reader::Reader;
+use std::collections::HashMap;
+use std::str;
+
+// Code doesn't work if DefaultSettings node is constructed like this:
+//   <DefaultSettings Language="es" Greeting="HELLO"/>
+
+const XML: &str = r#"
+<?xml version="1.0" encoding="utf-8"?>
+  <DefaultSettings Language="es" Greeting="HELLO">
+  </DefaultSettings>  
+  <Localization>
+    <Translation Tag="HELLO" Language="ja">
+        <Text>こんにちは</Text>
+    </Translation>
+    <Translation Tag="BYE" Language="ja">
+        <Text>さようなら</Text>
+    </Translation>
+     <Translation Tag="HELLO" Language="es">
+         <Text>Hola</Text>
+     </Translation>
+     <Translation Tag="BYE" Language="es">
+         <Text>Adiós</Text>
+     </Translation>
+  </Localization>
+"#;
+
+#[derive(Debug)]
+struct Translation {
+    tag: String,
+    lang: String,
+    text: String,
+}
+
+impl Translation {
+    fn new_from_element(
+        reader: &mut Reader<&[u8]>,
+        element: BytesStart,
+    ) -> Result<Translation, quick_xml::Error> {
+        use std::borrow::Cow;
+        let mut tag = Cow::Borrowed("");
+        let mut lang = Cow::Borrowed("");
+        // let (tag, lang) =
+        for attr_result in element.attributes() {
+            let a = attr_result.unwrap();
+            match a.key.as_ref() {
+                b"Language" => lang = a.unescape_value()?,
+                b"Tag" => tag = a.unescape_value()?,
+                _ => (),
+            }
+        }
+        let mut text = Cow::Borrowed("");
+        let mut element_buf = Vec::new();
+        let event = reader.read_event_into(&mut element_buf)?;
+        match event {
+            Event::Text(ref e) => {
+                text = str::from_utf8(e.as_ref())?.into();
+                println!("text node content: {}", text);
+            }
+            _ => (),
```

Comment:
Probably it would be better to return error here, because if you encounter, for example, a `Start` element here, then all further parsing will be broken.

You also can get an `Empty` or `End` event here, which corresponds to `<Text/>` and `<Text></Text>` pieces accordingly. I suppose you'd like to handle such situations correctly.

Maybe for an example this is not necessary, but at least a comment about that would be worth.

Translation also could be inside of CDATA section, so processing of `CData` event could be valuable.

Reply:
changed to use `reader.read_text` -- hoping that handles CDATA?

Reply:
No, it will return content of the source together with markup -- so for
```xml
<Text><![CDATA[cdata]]></Text>
```
it will return `<![CDATA[cdata]]>` instead of `cdata`

Nowadays there is not good method to do that in one-line, I'll plan to add such in #474. So for now manual processing is required.